### PR TITLE
Display the addon licenses from libzypp (bsc#1057223)

### PR DESCRIPTION
**Pull request by lslezak**
- [Trello card](https://trello.com/c/NFPbGpjO), [bsc#1057223](https://bugzilla.suse.com/show_bug.cgi?id=1057223)
- Added unit test
- Tested also manually using the latest build + updated `yast2-pkg-bindings` package

### Screenshot

Tested with the HA product from the Packages DVD:

addon_license_agreement2

### CODE REVIEW

File: src/modules/ProductLicense.rb [hunk 1]
```diff
@@ -1182,19 +1183,18 @@ def SearchForLicense_NormalRunBaseProduct(_src_id, fallback_dir)
     def SearchForLicense_AddOnProduct(src_id, _fallback_dir)
       Builtins.y2milestone("Getting license beta from repository %1", src_id)
 
+      # use a separate license directory for each product
+      @tmpdir = File.join(SCR.Read(path(".target.tmpdir")), "product-license", src_id.to_s)
+
+      # try reading the product license from libzypp first
+      return if product_license(src_id, @tmpdir)
```
Comment (jreidinger) [inline src/modules/ProductLicense.rb]: it will skip reading `@beta_file`, is it ok?

File: src/modules/ProductLicense.rb [hunk 2]
```diff
@@ -1624,6 +1624,59 @@ def beta_seen?(id)
...
+          log.info("License saved to #{path}")
+        end
+      end
+
+      if found_license
+        @license_dir = tmpdir
+        @license_file_print = repo_license_file(id)
+      end
+
+      found_license
```
Comment (jreidinger) [inline src/modules/ProductLicense.rb]: this means if we found license for at least one product, we are done with licenses?

File: src/modules/ProductLicense.rb (hunk 1 above)
Comment (lslezak) [reply src/modules/ProductLicense.rb]: Yes, that's OK. The Beta file cannot be placed into RPM-MD repositories, it's not accessible via libzypp from there.

File: src/modules/ProductLicense.rb [hunk 3]
```diff
@@ -1624,6 +1624,59 @@ def beta_seen?(id)
...
+      log.info("Found products from source #{id}: #{product_names.inspect}")
+      found_license = false
+
+      product_names.each do |product_name|
+        locales = Pkg.PrdLicenseLocales(product_name)
+        log.info("License locales for product #{product_name.inspect}: #{locales.inspect}")
+
+        locales.each do |locale|
+          license = Pkg.PrdGetLicenseToConfirm(product_name, locale)
+          next unless license && !license.empty?
```
[the remark this answers is not in the document]
Comment (lslezak) [reply src/modules/ProductLicense.rb]: Yeah, probably converted by Rubocop, I'll check it...

File: src/modules/ProductLicense.rb (hunk 2 above)
Comment (lslezak) [reply src/modules/ProductLicense.rb]: Yes, we can display license only for one product anyway. That's compatible with the old approach, there could be only one `license.tar.gz` file on the medium. No regression.

**Pull request merged**